editor/src/components/canvas/canvas-strategies/canvas-strategy-types.ts

```diff
@@ -49,13 +50,29 @@ export function strategyApplicationResult(
   }
 }
 
-export interface ControlWithKey {
-  control: React.FC<React.PropsWithChildren<unknown>>
+export interface ControlForStrategy<P> {
+  type: 'ControlForStrategy'
+  control: React.FC<P>
+}
+
+export function controlForStrategyMemoized<P>(control: React.FC<P>): ControlForStrategy<P> {
+  return { type: 'ControlForStrategy', control: React.memo<any>(control) }
+}
+
+export type WhenToShowControl =
+  | 'always-visible'
+  | 'visible-only-while-active'
+  | 'visible-except-when-other-strategy-is-active'
+
+export interface ControlWithProps<P> {
+  control: ControlForStrategy<P>
+  props: P
```

Comment:
maybe a silly question, but: given the usage (always `{}`), why aren't we making this optional and defaulting it to `{}` when using it in the memoized wrapper?
```ts
export interface ControlWithProps<P> {
  control: ControlForStrategy<P>
  props?: P
  ...
}
```

Reply:
that's actually a good question! I erred on the side that _for now_ these are {} but hopefully soon it will be filled up. 
but it would certainly make for a smaller diff if these would be optional...

Reply:
imho it would also make things less confusing: if I see code everywhere where `{}` is used, I have to wonder why that's the case 🤔 but of course maybe that's just me!

Reply:
oh you gave me a Typescript riddle!

if P is {} we want props to be optional. but if P is actually a non-empty props interface, we want to enforce props

Reply:
🤯

Reply:
I cannot come up with a good solution. But I promise that a follow-up PR is coming with non-empty props objects!